scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py

```diff
@@ -395,6 +432,106 @@ def Get(self, variable, flatten=True):
 
         return self.varData[variable][:]
 
+    def GetComponents(self, components, flatten=True):
+        """
+        Reads data for the named components from file.
+
+        Returns components as a dictionary of {component:component_data}
+
+        Throws an exception if the components do not exist in the file
+
+        Default is to return a flat array of length TotalCells.  However if
+        flatten is set to False, a 4D for each component array is returned that
+        has dimensions [NumBlocks, Nz, Ny, Nx] where NumBlocks is the total
+        number of blocks, and Nz, Ny, and Nx are the number of cells in the z,
+        y, and x directions respectively.
+        """
+
+        # Check if these components exist
+        missing_components = [
+            component
+            for component in components
+            if not component in self.ComponentsDatasetMap
+        ]
```

Comment:
I'm not familiar with this style of syntax, is this treating the stuff in brackets as a single line? Otherwise my intuition tells me the if statement is missing an indentation?

Reply:
I believe that inside a list, you can have line breaks for free. @forrestglines is using a list comprehension here to build the `missing_components` list on the fly.

Reply:
Yep, line breaks work across any bracketing ie. `( ) [ ] { }`. This is just a list comprehension don't remember formatting like this - the python formatter may have done it.